### PR TITLE
app-layer-imap: Add IMAP support.

### PR DESCRIPTION
**Pull request by decanio**
Add IMAP application layer parser and eve-log JSON logger.

https://buildbot.openinfosecfoundation.org/builders/decanio/builds/28
https://buildbot.openinfosecfoundation.org/builders/decanio-pcap/builds/28

### CODE REVIEW

File: src/app-layer-imap.c [hunk 1]
```diff
@@ -0,0 +1,1403 @@
...
+            }
+        }
+        files = imap_state->files_ts;
+
+        /* Open file if necessary */
+        if (state->body_begin) {
+
+            if (SCLogDebugEnabled()) {
+                SCLogDebug("Opening file...%u bytes", len);
+                printf("File - ");
```
Comment (gozzy) [inline src/app-layer-imap.c]: 3 printfs look like a debug. And filename printing looks a bit strange.

Comment (decanio) [inline src/app-layer-imap.c]: Yes.  This was put in during initial development.  Its served its purpose.  I'll pull it out.

**Comment by inliniac** [comment]
Can you add AFL entry point commandline options. See https://github.com/inliniac/suricata/blob/master/src/suricata.c#L1396

**Comment by decanio** [comment]
I'm happy to add support for AFL to this.  Do you want this as a part of this PR or a separate one that gets applied on top of this one?  Remember I'm already planning on sending you additional PRs that add "filedata" and LUA support to IMAP.

**Comment by inliniac** [comment]
I'd like to run the AFL test before merging anything, so as part of the PR would be great :)

**Pull request closed without merging**